File: Application/Sources/Helpers/Categories/NSArray+PlaySRG.m
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

#import "NSArray+PlaySRG.h"

@implementation NSArray (PlaySRG)

- (NSArray *)play_arrayByRemovingObjectAtIndex:(NSUInteger)index
{
    NSMutableArray *mutableArray = self.mutableCopy;
    [mutableArray removeObjectAtIndex:index];
    return mutableArray.copy;
}

- (NSArray *)play_arrayByRemovingObjectsInArray:(NSArray *)array
{
    NSMutableArray *mutableArray = self.mutableCopy;
    [mutableArray removeObjectsInArray:array];
    return mutableArray.copy;
}

@end

File: Application/Sources/Helpers/Extensions/NSArray+PlaySRG.swift
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

import Foundation

extension NSArray {
    /**
     Returns a new array which is the receiver with objects from the specified array removed.
     
     - parameter array: The array of objects to remove from the receiver.
     
     - returns: A new array with objects from the specified array removed.
     */
    @objc func arrayByRemovingObjects(in array: [AnyHashable]) -> NSArray {
        let mutableArray = NSMutableArray(array: self)
        mutableArray.removeObjects(in: array)
        return NSArray(array: mutableArray)
    }
}
